src/ucp/core/ucp_mm.c

```diff
@@ -1385,36 +1400,38 @@ ucs_status_t ucp_mem_reg_md_map_update(ucp_context_h context)
     context->alloc_md_index[alloc_mem_type] = alloc_memh->alloc_md_index;
     ucp_memh_put(context, alloc_memh);
 
+    return UCS_OK;
+}
+
+ucs_status_t
+ucp_mm_get_reg_md_map(ucp_context_h context, ucp_md_map_t *md_map_p)
+{
+    const ucs_memory_type_t reg_mem_type = UCS_MEMORY_TYPE_HOST;
+    ucp_mem_dummy_handle_t reg_memh      = {
+        .memh.alloc_md_index = UCP_NULL_RESOURCE
+    };
+    ucs_status_t status;
+    uint8_t buff;
+
+    if (context->alloc_md_map_initialized) {
+        goto ret;
+    }
+
     status = ucp_memh_register(context, &reg_memh.memh,
-                               context->reg_md_map[alloc_mem_type], &buff,
-                               sizeof(buff), alloc_mem_type,
+                               context->reg_md_map[reg_mem_type], &buff,
+                               sizeof(buff), reg_mem_type,
                                UCT_MD_MEM_ACCESS_ALL |
-                               UCT_MD_MEM_FLAG_HIDE_ERRORS);
+                                       UCT_MD_MEM_FLAG_HIDE_ERRORS);
     if (status != UCS_OK) {
         return status;
     }
 
-    context->reg_md_map[alloc_mem_type] = reg_memh.memh.md_map;
+    context->alloc_md_map_initialized = 1;
+    context->alloc_md_map             = reg_memh.memh.md_map;
     ucp_memh_dereg(context, &reg_memh.memh, reg_memh.memh.md_map);
 
-    /* If we have a dmabuf provider for a memory type, it means we can register
-     * memory of this type with any md that supports dmabuf registration. */
-    ucs_memory_type_for_each(mem_type) {
-        if (context->dmabuf_mds[mem_type] != UCP_NULL_RESOURCE) {
-            context->reg_md_map[mem_type] |= context->dmabuf_reg_md_map;
-        }
-
-        ucs_string_buffer_reset(&strb);
-        ucs_for_each_bit(md_index, context->reg_md_map[mem_type]) {
-            ucs_string_buffer_appendf(&strb, "%s, ",
-                                      context->tl_mds[md_index].rsc.md_name);
-        }
-        ucs_string_buffer_rtrim(&strb, ", ");
-
-        ucs_debug("register %s memory on: %s", ucs_memory_type_names[mem_type],
-                  ucs_string_buffer_cstr(&strb));
-    }
-
+ret:
+    *md_map_p = context->alloc_md_map;
```

Comment:
it is not the same, alloc_md_map may contain some MDs where registration is not supported (e.g. sysv)

Reply:
Guess the name is a little confusing.
There are `alloc_md_map`, `reg_md_map` and `alloc_md_index`.

Reply:
so, the alloc_md_map is not the same as reg_md_map

Reply:
Originally this code was added here https://github.com/openucx/ucx/pull/7354

Reply:
Removed.